src/address.rs

```diff
@@ -1,116 +1,90 @@
 use crate::error::Error;
 use crate::SubnetID;
+use fil_actors_runtime::cbor;
+use fvm_ipld_encoding::RawBytes;
 use fvm_shared::address::Address;
 use fvm_shared::ActorID;
 use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
-use std::fmt;
-use std::fmt::Formatter;
 use std::str::FromStr;
 
-/// Maximum length for an address payload determined by
-/// the maximum size of the IPC address.
-const MAX_ADDRESS_LEN: usize = 54;
-
 // The default actor id namespace for IPC addresses
 lazy_static! {
     pub static ref DEFAULT_IPC_ACTOR_NAMESPACE_ID: ActorID = 1000u64;
 }
 
 #[derive(Clone, PartialEq, Eq, Debug, Hash, Serialize, Deserialize)]
 pub struct IPCAddress {
-    inner: Address,
-}
-
-impl fmt::Display for IPCAddress {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        self.inner.fmt(f)
-    }
+    subnet_id: SubnetID,
+    raw_address: Address,
 }
 
 impl IPCAddress {
     /// Generates new address using ID protocol
-    pub const fn new_id(id: u64) -> Self {
+    pub fn new_id(id: u64) -> Self {
+        let d = SubnetID::default();
```

Comment:
`SubnetID::default()` doesn't seem to represent a meaningful subnet identifier. How would it be interpreted? Should we use `Option` in case of a relative IPC address?

Reply:
Not sure if there exists a sensible default value for `SubnetID`. @adlrocha WDYT?

Reply:
```
impl Default for SubnetID {
    fn default() -> Self {
        Self {
            parent: String::from(""),
            actor: Address::new_id(0),
        }
    }
}
```

Reply:
Then, if we are removing `Result` from some of the IPC methods, let's make the rootnet the default `SubnetID`. Thanks!

Reply:
Do you think the root is a sensible default value? Otherwise, I'd remove the implementation of `Default`.

Reply:
> Should we use `Option` in case of a relative IPC address?

I meant defining `IPCAddress` as follows

```rust
 pub struct IPCAddress {
    subnet_id: Option<SubnetID>,
    local_addr: Address,
}
```

where `subnet_id == None` signifies IPC address relative to the current subnet.

Reply:
Or we better explicitly support relative and absolute `SubnetID`s and let `SubnetID::default()` represent something like `.` directory name in POSIX.

Reply:
maybe we use `Result` for now as api? Once in prod, we might need to support multiple versions of `IPCAddress` and `Result` should be more general as compared?

Reply:
Oh yes, you're right!